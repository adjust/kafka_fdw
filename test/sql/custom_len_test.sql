\i test/sql/setup.inc
CREATE FOREIGN TABLE kafka_test_custom (
    part int OPTIONS (partition 'true'),
    offs bigint OPTIONS (offset 'true'),
    len int
)
SERVER kafka_server OPTIONS
    (format 'custom', decode_lib '$libdir/kafka_fdw', decode_func 'kafka_custom_msglen' ,topic 'contrib_regress', batch_size '30', buffer_delay '500');

SELECT part, offs, len  FROM kafka_test_custom WHERE offs BETWEEN 1 AND 5 ORDER BY offs;