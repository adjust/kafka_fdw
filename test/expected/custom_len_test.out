\i test/sql/setup.inc
\set ECHO none
CREATE FOREIGN TABLE kafka_test_custom (
    part int OPTIONS (partition 'true'),
    offs bigint OPTIONS (offset 'true'),
    len int
)
SERVER kafka_server OPTIONS
    (format 'custom', decode_lib '$libdir/kafka_fdw', decode_func 'kafka_custom_msglen' ,topic 'contrib_regress', batch_size '30', buffer_delay '500');
SELECT part, offs, len  FROM kafka_test_custom WHERE offs BETWEEN 1 AND 5 ORDER BY offs;
 part | offs | len 
------+------+-----
    0 |    1 |  73
    0 |    2 |  73
    0 |    3 |  73
    0 |    4 |  73
    0 |    5 |  73
(5 rows)

